Library/ViewModels/SignupViewModel.swift

```diff
@@ -4,108 +4,85 @@ import Prelude
-  /// Sets whether the email text field is the first responder.
-  var emailTextFieldBecomeFirstResponder: Signal<(), NoError> { get }
-
-  /// Emits true when the signup button should be enabled, false otherwise.
-  var isSignupButtonEnabled: Signal<Bool, NoError> { get }
-
-  /// Emits an access token envelope that can be used to update the environment.
-  var logIntoEnvironment: Signal<AccessTokenEnvelope, NoError> { get }
-
-  /// Sets whether the password text field is the first responder.
-  var passwordTextFieldBecomeFirstResponder: Signal<(), NoError> { get }
-
-  /// Emits when a notification should be posted.
-  var postNotification: Signal<Notification, NoError> { get }
-
-  /// Sets whether the name text field is the first responder.
-  var nameTextFieldBecomeFirstResponder: Signal<(), NoError> { get }
-
-  /// Emits the value for the weekly newsletter.
-  var setWeeklyNewsletterState: Signal<Bool, NoError> { get }
-
-  /// Emits when a signup error has occurred and a message should be displayed.
-  var showError: Signal<String, NoError> { get }
-}
-
-public protocol SignupViewModelType {
-  var inputs: SignupViewModelInputs { get }
-  var outputs: SignupViewModelOutputs { get }
-}
-
-public final class SignupViewModel: SignupViewModelType, SignupViewModelInputs, SignupViewModelOutputs {
+public final class SignupViewModel {
+  public struct Inputs {
+    /// Call when the user enters a new email address.
+    public let (emailTextChangedSignal, emailTextChangedObserver) = Signal<String?, NoError>.pipe()
+    /// Call when the user returns from email text field.
+    public let (emailTextFieldDidReturnSignal, emailTextFieldDidReturnObserver) = Signal<(), NoError>.pipe()
+    /// Call when the environment has been logged into
+    public let (environmentLoggedInSignal, environmentLoggedInObserver) = Signal<(), NoError>.pipe()
+    /// Call when the user enters a new name.
+    public let (nameTextChangedSignal, nameTextChangedObserver) = Signal<String?, NoError>.pipe()
+    /// Call when the user returns from the name text field.
+    public let (nameTextFieldDidReturnSignal, nameTextFieldDidReturnObserver) = Signal<(), NoError>.pipe()
+    /// Call when the user enters a new password.
+    public let (passwordTextChangedSignal, passwordTextChangedObserver) = Signal<String?, NoError>.pipe()
+    /// Call when the user returns from the password text field.
+    public let (passwordTextFieldDidReturnSignal, passwordTextFieldDidReturnObserver) =
+      Signal<(), NoError>.pipe()
+    /// Call when the user taps signup.
+    public let (signupButtonPressedSignal, signupButtonPressedObserver) = Signal<(), NoError>.pipe()
+    /// Call when the view did load.
+    public let (viewDidLoadSignal, viewDidLoadObserver) = Signal<(), NoError>.pipe()
+    /// Call when the user toggles weekly newsletter.
+    public let (weeklyNewsletterChangedSignal, weeklyNewsletterChangedObserver) = Signal<Bool, NoError>.pipe()
+
+    init() {}
```

Comment:
Strange that Swift wouldn't let us get away without this when we use the `pipe()`s!